site/source/docs/executing_scripts.md

```diff
@@ -0,0 +1,115 @@
+
+It's important to note that `afterDeploy` hooks are executed every time all Smart Contracts have been deployed. Often there are cases where running a (migration) script manually is what you really need.
+
+Scripts let you exactly that as they can be run at any time, regardless of what your app's current deployment status is.
+
+## What's a script?
+
+A script is really just a file with an exported function that has special dependencies injected into it. Here's what it could look like:
+
+```
+modules.exports = async ({ contracts, web3, logger}) => {
+  ...
+};
+```
+
+The injected parameters are:
+
+- `contracts` - A map object containing all of your Smart Contracts as Embark Smart Contract instances.
+- `web3` - A web3 instances to give you access to things like accounts.
+- `logger` - Embark's custom logger.
+
+Scripts can be located anywhere on your machine but should most likely live inside your project's file tree in a dedicated folder.
+
+## Running scripts
+
+To run a script, use the CLI `exec` command and specify an environment as well as the script to be executed:
+
+```
+$ embark exec development scripts/001.js
+```
+
+The command above will execute the function in `scripts/001.js` and ensures that Smart Contracts are deployed in the `development` environment.
+
+If you have multiple scripts that should run in order it's also possible to specify the directory in which they live in:
+
+```
+$ embark exec development scripts
+```
+
+Embark will then find all script files inside the specified directory (in this case `scripts`) and then run them one by one. If any of the scripts fails by emitting an error, Embark will abort the execution. Scripts are executed in sequence which means, all following scripts won't be executed in case of an error.
+
+## Error Handling
+
+It's possible and recommended for scripts to emit proper errors in case they fail to do their job. There are several ways to emit an error depending on how you write your function. Scripts are executed asyncronously, so one way to emit an error is to reject a promise:
+
+```
+modules.exports = () => {
+  return new Promise((resolve, reject) => {
+    reject(new Error('Whoops, something went wrong'));
+  });
+};
+
+// or
+modules.exports = () => {
+  return Promise.reject(new Error ('Whoops, something went wrong'));
+};
+```
+
+If your script uses the `async/await` syntax, errors can also be emitted by using other `async` APIs that fail:
+
+```
+module.exports = async () => {
+  await someAPIThatFails(); // this will mit an error
+};
+```
+
+If an error is emitted Embark will do its best to give you information about the original error:
+
+```
+.. 001.js running....
+Script '001.js' failed to execute. Original error: Error: Whoops, something went wrong
```

Comment:
We should probably give a stack trace too. I know it's a code change, but I just noticed here that we don,t give the trace

Reply:
That's a very good suggestion, thanks!